lib/facter/rabbitmq_plugins_dirs.rb

```diff
@@ -2,7 +2,8 @@
   setcode do
     if Facter::Util::Resolution.which('rabbitmqctl')
       rabbitmq_pluginsdirs_env = Facter::Core::Execution.execute("rabbitmqctl eval 'application:get_env(rabbit, plugins_dir).'")
-      rabbitmq_plugins_dirs = %r{^\{ok\,\"(\/.+\/\w+)}.match(rabbitmq_pluginsdirs_env)[1]
+      rabbitmq_plugins_dirs_match = %r{^\{ok\,\"(\/.+\/\w+)}.match(rabbitmq_pluginsdirs_env)
+      rabbitmq_plugins_dirs = rabbitmq_plugins_dirs_match ? rabbitmq_plugins_dirs_match[1] : ''
```

Comment:
This looks basically right, I wonder if it can / should be nil vs an empty string? Also maybe not using the ternary operator would be a little easier to read?

Reply:
Thanks for the review :).

Re nil vs. string: i think it should be an empty string, because on the line below we call .split(':') on it, which produces an empty array as the final fact value, which i think is correct. `nil` would crash on the `split` call.

Re ternary: I can change it, what would be your preference? Something like this?
```ruby
rabbitmq_plugins_dirs = if rabbitmq_plugins_dirs_match then rabbitmq_plugins_dirs_match[1] else '' end
```

Reply:
I think that would look good, maybe not all on the same line?

I'm pretty sure the fact itself can return an empty string, so can we return `nil` _after_ trying to split?

https://puppet.com/docs/facter/3.6/fact_overview.html

Reply:
Ok i see, previously i didn't understand that you suggested to return `nil` instead of `[]` from the fact block itself. I can make it so :). I'll amend the patch for another review round.